client-lite/src/download/download_manager.cpp

```diff
@@ -102,12 +103,20 @@ DownloadStatus DownloadManager::GetDownloadStatus(const std::string& downloadId)
 bool DownloadManager::IsIdle() const
 {
     // Reset _fRunning if we are idle to disallow new downloads.
-    // This handles the race between shutdown and new download request coming in.
+    // This handles the race between shutdown or policy refresh and new download request coming in.
     std::unique_lock<std::shared_timed_mutex> lock(_downloadsMtx);
     _fRunning = !_downloads.empty();
     return !_fRunning;
 }
 
+void DownloadManager::RefreshAdminConfigs() const
+{
+    _taskThread.SchedImmediate([&]()
```

Comment:
>& [](start = 32, length = 1)

Capture 'this' instead. The capture-all option is generally discouraged because it will capture more stuff than is needed. There's nothing other than 'this' to capture here but better to get used to that practice.